wire-ios-data-model/Source/MLS/OneOnOne/OneOnOneResolver.swift

```diff
@@ -92,68 +91,151 @@ public final class OneOnOneResolver: OneOnOneResolverInterface {
     ) async throws -> OneOnOneConversationResolution {
         WireLogger.conversation.debug("resolving 1-1 conversation with user: \(userID)")
 
-        let messageProtocol = try await protocolSelector.getProtocolForUser(
-            with: userID,
-            in: context
-        )
+        let messageProtocol = try await protocolSelector.getProtocolForUser(with: userID, in: context)
 
         switch messageProtocol {
         case .none:
-            WireLogger.conversation.debug("no common protocols found")
-            await context.perform {
-                guard
-                    let otherUser = ZMUser.fetch(with: userID, in: context),
-                    let conversation = otherUser.oneOnOneConversation
-                else {
-                    return
-                }
+            return await resolveCommonUserProtocolNone(with: userID, in: context)
+        case .mls:
+            return try await resolveCommonUserProtocolMLS(with: userID, in: context)
+        case .proteus:
+            return resolveCommonUserProtocolProteus()
+        case .mixed:
+            return resolveCommonUserProtocolMixed()
```

Comment:
This can never happen a user cannot support `.mixed` as a protocol. Only conversations have the protocol set to `.mixed`.

Reply:
My idea was to have the same level of abstraction in the switch, but I can change it back :)

<img width="960" alt="Screenshot 2024-03-21 at 12 47 42" src="https://github.com/wireapp/wire-ios/assets/7196958/35afd45e-2aae-4c79-a260-22d3993a84ac">